File: frontends/nextjs/src/lib/db/functions/tenants/delete-tenant.ts
import { prisma } from '../../prisma'

/**
 * Delete a tenant by ID
 * @param tenantId - The tenant ID
 */
export const deleteTenant = async (tenantId: string): Promise<void> => {
  await prisma.tenant.delete({ where: { id: tenantId } })
}

File: frontends/nextjs/public/packages/ui_level5/seed/scripts/tenants/delete_tenant.lua
-- Delete tenant action handler
-- Single function module for tenant management

---@class DeleteTenant
local M = {}

---Handle delete tenant action
---@param ctx TenantContext Context with tenantId
---@return ActionResult
function M.deleteTenant(ctx)
  return { success = true, action = "confirm_delete", id = ctx.tenantId }
end

return M
